api/views.py

```diff
@@ -0,0 +1,49 @@
+from django.utils.timezone import datetime
+import datetime
+from rest_framework.response import Response
+from rest_framework.viewsets import ModelViewSet
+from rest_framework import status
+
+from api.models import Job, Application
+from api.serializers import JobSerializer, ApplicationSerializer
+from api.common.constants import DATA_SAVED_SUCCESSFULLY, DATA_ALREADY_EXISTS
+
+
+# Create your views here.
+class JobViewSet(ModelViewSet):
+    queryset = Job.objects.all().filter(is_active=True)
+    serializer_class = JobSerializer
+
+    def list(self, request):
+        category = request.GET.get('category')
+
+        if category:
+            jobs = self.queryset.filter(category_job__id=category)
+        else:
+            jobs = self.queryset.filter()
+
+        job = jobs.filter(expiration_date__date__lt=datetime.date.today())
+        job.update(is_active=False)
+
+        serializer = self.serializer_class(jobs.filter(is_active=True), many=True)
+        return Response(data=serializer.data, status=status.HTTP_200_OK)
+
+
+class ApplicationViewSet(ModelViewSet):
+    queryset = Application.objects.all()
+    serializer_class = ApplicationSerializer
+
+    def create(self, request):
+        job_id = request.data['applicant_application']
+        user_email = request.data['email']
+        application = self.queryset.filter(applicant_application_id=job_id, email=user_email).first()
+
+        if not application:
+            serializer = self.serializer_class(data=request.data)
+
+            if serializer.is_valid():
+                serializer.save()
+                return Response(data=DATA_SAVED_SUCCESSFULLY, status=status.HTTP_201_CREATED)
+            return Response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(data=DATA_ALREADY_EXISTS, status=status.HTTP_400_BAD_REQUEST)
```

Comment:
@amina-cg wrong use of conditions. Always try to make less block of conditions.
```suggestion
        if application:
            Response(data=DATA_ALREADY_EXISTS, status=status.HTTP_400_BAD_REQUEST)
        serializer = self.serializer_class(data=request.data)
        if not serializer.is_valid():
            return Response(data=serializer.errors, status=status.HTTP_400_BAD_REQUEST)
        serializer.save()
        return Response(data=DATA_SAVED_SUCCESSFULLY, status=status.HTTP_201_CREATED)
```

Reply:
Always try to use the `early return` approach.*

A wise man once said 
"The faster you can fail the faster you can succeed."